isis/src/base/apps/csminit/csminit.cpp

```diff
@@ -411,7 +410,10 @@ namespace Isis {
     cube->deleteBlob("Polygon", "Footprint");
 
     // Create our CSM State blob as a string and add the CSM string to the Blob.
-    StringBlob csmStateBlob(modelState, "CSMState");
+    Blob csmStateBlob("CSMState", "String");
+    char *buf = new char[modelState.size()];
+    memcpy(buf, modelState.c_str(), modelState.size());
+    csmStateBlob.setData(buf, modelState.size()); // This passes ownership so don't delete buf
```

Comment:
We could do this in the set data function, but then we would need to delete the locally scoped char blob. Maybe we could also do a blob constructor from a string?

Reply:
Yeah, I had a hard time figuring out a good way to do this. We could have Blob::setData do this and change it to take a const char* I'll look and see if that works better.

Reply:
This is something that I started to comment on in a review as well, but I wasn't really sure where a function like this would live.  I think changing setData makes the most sense

Reply:
I ended up having 2 ways to set that data. You can call setData to copy the data and takeData to pass the data. This is because of how Table stores it data. With the copy inside of setData you end up with 3 copies of the Table data at one point.